cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/service/ValidationHandler.java

```diff
@@ -82,16 +87,36 @@ public void healthCheck(HttpServiceRequest request, HttpServiceResponder respond
   @Path("v1/contexts/{context}/validations/stage")
   public void validateStage(HttpServiceRequest request, HttpServiceResponder responder,
                             @PathParam("context") String namespace) throws IOException, AccessException {
-    validateLocally(request, responder, namespace);
-  }
-
-  private void validateLocally(HttpServiceRequest request, HttpServiceResponder responder,
-                               String namespace) throws IOException {
     if (!getContext().getAdmin().namespaceExists(namespace)) {
       responder.sendError(HttpURLConnection.HTTP_NOT_FOUND, String.format("Namespace '%s' does not exist", namespace));
       return;
     }
 
+    //Validate remotely if remote execution is enabled
+    if (getContext().remoteExecutionEnabled()) {
```

Comment:
this should come from cconf.

Reply:
There is no access to cconf in ValidationHandler (since it is a system app) and the required context is provided with SystemHttpServiceContext. The implementation of this method reads from cconf.